rusoto/core/src/xmlutil.rs

```diff
@@ -168,6 +168,23 @@ pub fn skip_tree<T: Peek + Next>(stack: &mut T) {
     }
 
 }
+
+/// skip all elements until a start element is encountered
+///
+/// Errors and end-of-stream are ignored.
+pub fn find_start_element<T: Peek + Next>(stack: &mut T) {
+    loop {
+        match stack.peek() {
+            Some(&Ok(XmlEvent::StartElement { .. })) => break,
+            Some(&Ok(_)) => {
+                stack.next().unwrap().unwrap();
```

Comment:
These unwraps seems a little spooky to me. Mainly because I'm not to familiar with our XML Parsing. Can you explain to me why these two unwraps are safe, and what circumstances it'd be triggered under?

Reply:
Unless there is something majorly wrong, the `unwrap()`s should never panic because we just `peek()`ed at what's coming `next()` and we know it is `Some(&Ok(_))`.

Reply:
Yea that's totally right. I really should of double checked that instead of just a quick glance through.